// engine/index.js

var Utils = require('../../utils'),

    Math = require('../../client/engine/utils/Math'),
    RandomGenerator = require('../../client/engine/utils/RandomGenerator'),
    Utility = require('../../client/engine/utils/Utility'),
    
    Point = require('../../client/engine/geometry/Point'),
    Line = require('../../client/engine/geometry/Line'),
    Rectangle = require('../../client/engine/geometry/Rectangle'),
    Circle = require('../../client/engine/geometry/Circle'),
    Ellipse = require('../../client/engine/geometry/Ellipse'),
    
    Easing = require('../../client/engine/tween/Easing'),
    Timer = require('../../client/engine/time/Timer');

module.exports =
  Utils.extend(require('../../client/engine/const'), {
    // geometry
    Point: Point,
    Line: Line,
    Rectangle: Rectangle,
    Circle: Circle,
    Ellipse: Ellipse,

    // easing and timer
    Timer: Timer,
    Easing: Easing,

    // utils
    Math: Math,
    RandomGenerator: RandomGenerator,
    Utility: Utility
  });
